core/src/main/resources/hudson/model/DirectoryBrowserSupport/dir.jelly

```diff
@@ -114,7 +114,7 @@ THE SOFTWARE.
               <tr>
                 <td style="text-align:right;" colspan="3">
                   <div style="margin-top: 1em;">
-                    <a class="jenkins-link--with-icon" href="${backPath}${pattern!=''?pattern+'/':''}*zip*/${dir.name}.zip">
+                    <a class="jenkins-link--with-icon" href="${backPath}${pattern!=''?pattern+'/':''}*zip*/${h.urlEncode(dir.name)}.zip">
```

Comment:
Do spaces in the directories still work?

We had a similar issue in the junit plugin and `rawEncode` had to be used instead of `rawEncode`

https://github.com/jenkinsci/junit-plugin/pull/660#issuecomment-2465965560

https://github.com/jenkinsci/junit-plugin/pull/661

and finally:
https://github.com/jenkinsci/junit-plugin/pull/668